definitions/procedures/foreman_tasks/cleanup.rb

```diff
@@ -0,0 +1,53 @@
+module Procedures::ForemanTasks
+  class Cleanup < ForemanMaintain::Procedure
+    PERMITTED_STATES = %w[all pending scheduled planning planned running paused stopped].freeze
+    metadata do
+      for_feature :foreman_tasks
+      description 'Perform task cleanup'
+      preparation_steps { Checks::Foreman::DBUp.new }
+      confine do
+        feature(:foreman) && feature(:foreman_tasks)
+      end
+
+      param :states,
+            'Operate on tasks in STATES' \
+             "\nAvailable states: " + PERMITTED_STATES.join(', '),
+            :array => true, :allowed_values => PERMITTED_STATES
+      param :batch_size,
+            'Process tasks in batches of BATCH_SIZE, 1000 by default' do |s|
+        Integer(s) unless s.nil?
+      end
+      param :after,
+            'Operate on tasks older than AFTER. Expected format is a number ' \
+            "followed by the time unit (s,h,d,m,y), such as '10d' for 10 days"
+      param :backup, 'Backup deleted tasks', :flag => true
```

Comment:
I tried to pass the backup option but it did not work. I also tried to change the value of `TASK_BACKUP` to `true`instead of `1` but no effect. Request to check this one.

Reply:
Should be fixed now